pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
 		e.Opts.IsEnabled(OptionEgressPolicy)
 }
 
+func (e *Endpoint) GetPortNumber(port string) (uint16, error) {
+	i, err := strconv.Atoi(port)
+	if err != nil {
+		// FIXME check for ports names here if it doesn't exist, the throw initial error
+		return 0, err
+	}
+	return uint16(i), nil
+}
+
+func (e *Endpoint) regeneratePolicyNew(owner Owner, opts models.ConfigurationMap) (*policy.Policy, *policy.Policy, uint64) {
```

Comment:
This function should not be in the endpoint package. We should define an interface for what the policy package needs from the endpoint.

Reply:
Oh, because the policy package will be the one calling the endpoint to regenerate its policy?

Reply:
The policy package should no longer be tied to the endpoint. What if we want policy on other things than endpoints in the future?

The policy package needs certain properties from whatever it specifies policy for. Right now these are the labels/identity and the policy specific configuration. We should list all of these requirements in an interface and then move all policy specific code into `pkg/policy` or a sub-package thereof.